slatedb/src/wal_buffer.rs

```diff
@@ -69,7 +74,7 @@ struct WalBufferManagerInner {
     /// The flusher will try flush all the immutable wals to remote storage.
     immutable_wals: VecDeque<(u64, Arc<KVTable>)>,
     /// The channel to send the flush work to the background worker.
-    flush_tx: Option<mpsc::Sender<WalFlushWork>>,
+    flush_tx: Option<mpsc::UnboundedSender<WalFlushWork>>,
```

Comment:
This is now unbounded. I think this means we could run out of memory if a user calls flush over and over again. But that's true for the mem_table_flusher as well, I believe.

We could check `flush_rx.len()` if we want to drop flush calls when a certain size is exceeded. Unfortunately, we can't easily swap bounded for unbounded because tokio has two entirely different channel types with no common trait.